frame/contracts/src/benchmarking.rs

```diff
@@ -122,69 +448,62 @@ benchmarks! {
 
 	// This constructs a contract that is maximal expensive to instrument.
 	// It creates a maximum number of metering blocks per byte.
+	// `n`: Size of the code in kilobytes.
 	put_code {
-		let n in 0 .. Contracts::<T>::current_schedule().max_code_size;
+		let n in 0 .. Contracts::<T>::current_schedule().max_code_size / 1024;
 		let caller = create_funded_user::<T>("caller", 0);
-		let (binary, hash) = expanded_contract::<T>(n);
-	}: _(RawOrigin::Signed(caller), binary)
+		let module = sized_code::<T>(n * 1024);
+		let origin = RawOrigin::Signed(caller);
```

Comment:
Can you elaborate? What does it mean for a caller to be 'whitelisted' and why do you think this is appropriate here?

Reply:
> What does it mean for a caller to be 'whitelisted'

usually in pallets dispatchables we whitelist the AccountInfo of the caller. I.e. the value for the caller account in the storage map `frame_system::AccountInfo` is counted as a read or a write in benchmarks. This is because the account info is already read and write for paying the fees of the extrinsic, thus it is part of ExtrinsicBaseWeight.

> why do you think this is appropriate here?

Actually I'm not sure it is appropriate anymore. Because you don't make use of ExtrinsicBaseWeight when you compute the gas of a contract calling another contract or stuff like this.

So it seems fine looking again now

Reply:
Thanks for the explanation. I think for the five **extrinsic** benchmarks we should whitelist the caller as you suggested initially.  Those are pretty much like any other extrinsic benchmark.

However, for the API benchmarks, you are right to say that we should not whitelist there.

Reply:
yes maybe, anyway be careful not to end up in some miss like https://github.com/paritytech/substrate/pull/7196

Reply:
Right! I think we are in the clear hear as the contract extrinsics do not make use of any inner call. I updated those benchmarks to whitelist the call.